application/app/src/main/java/uk/nhs/digital/uec/api/service/impl/FuzzyServiceSearchService.java

```diff
@@ -134,71 +115,71 @@ public List<DosService> retrieveServicesByGeoLocation(
-      /** Call the auth service login endpoint from here and get the authenticated headers */
-      MultiValueMap<String, String> headers = externalApiHandshakeInterface.getAccessTokenHeader();
-      log.info("Calling Postcode API to get location values");
-      /** Calculate distance to services returned if we have a search location */
-      PostcodeLocation searchLocation =
-        locationService.getLocationForPostcode(searchPostcode, headers);
-
-      List<PostcodeLocation> dosServicePostCodeLocation =
-        this.populateEmptyLocation(nonPopulatedLatLongServices, headers);
-      if (searchLocation != null) {
-        for (DosService dosService : nonPopulatedLatLongServices) {
-          PostcodeLocation serviceLocation = new PostcodeLocation();
-          serviceLocation.setPostcode(dosService.getPostcode());
-          serviceLocation.setEasting(dosService.getEasting());
-          serviceLocation.setNorthing(dosService.getNorthing());
-
-          if (serviceLocation.getEasting() == null && serviceLocation.getNorthing() == null) {
-            setServiceLocation(serviceLocation, dosServicePostCodeLocation);
-          }
-          dosService.setDistance(locationService.distanceBetween(searchLocation, serviceLocation));
-        }
-      }
-    }
-    return nonPopulatedLatLongServices;
-  }
-
-  private void setServiceLocation(
-    PostcodeLocation serviceLocation, List<PostcodeLocation> dosServicePostCodeLocation) {
-    if (dosServicePostCodeLocation != null) {
-      String servicePostcodeWithoutSpace =
-        apiUtilsService.removeBlankSpaces(serviceLocation.getPostcode());
-      serviceLocation.setEasting(
-        dosServicePostCodeLocation.stream()
-          .filter(t -> t.getPostcode().equals(servicePostcodeWithoutSpace))
-          .map(PostcodeLocation::getEasting)
-          .findFirst()
-          .orElse(null));
-      serviceLocation.setNorthing(
-        dosServicePostCodeLocation.stream()
-          .filter(t -> t.getPostcode().equals(servicePostcodeWithoutSpace))
-          .map(PostcodeLocation::getNorthing)
-          .findFirst()
-          .orElse(null));
-    }
-  }
-
-  private List<PostcodeLocation> populateEmptyLocation(
-    List<DosService> dosServices, MultiValueMap<String, String> headers)
-    throws InvalidParameterException {
-    log.info("Populating empty location on service");
-    List<String> postCodes =
-      dosServices.stream()
-        .filter(t -> t.getEasting() == null && t.getNorthing() == null)
-        .map(DosService::getPostcode)
-        .collect(Collectors.toList());
-    return !CollectionUtils.isNullOrEmpty(postCodes)
-      ? locationService.getLocationsForPostcodes(postCodes, headers)
-      : Collections.emptyList();
-  }
+//  private List<DosService> populateServiceDistancesWithNorthingAndEastings(List<DosService> nonPopulatedLatLongServices, String searchPostcode) throws InvalidParameterException, NotFoundException {
```

Comment:
I would suggest implementing the google api to get the lat and long from the user request so we can delete this logic

Reply:
I see where you are coming from with regard to empty locations. I will integrate google API